src/DurableTask.AzureStorage/AzureStorageOrchestrationService.cs

```diff
@@ -1857,8 +1856,16 @@ async Task<PurgeResult> IOrchestrationServicePurgeClient.PurgeInstanceStateAsync
                 throw new ArgumentException(nameof(instanceId));
             }
 
+            bool isInfiniteTimeSpan = timeout == Timeout.InfiniteTimeSpan;
+            if (timeout < TimeSpan.Zero && !isInfiniteTimeSpan)
+            {
+                throw new ArgumentException($"The parameter {nameof(timeout)} cannot be negative." +
+                    $" The value for {nameof(timeout)} was '{timeout}'." +
+                    $" Please provide either positive timeout value, or `Timeout.InfiniteTimeSpan`.");
+            }
+
             TimeSpan statusPollingInterval = TimeSpan.FromSeconds(2);
-            while (!cancellationToken.IsCancellationRequested && timeout > TimeSpan.Zero)
+            while (!cancellationToken.IsCancellationRequested && (isInfiniteTimeSpan || (timeout > TimeSpan.Zero)))
```

Comment:
I wonder if we should change this to support `TimeSpan.Zero`.
```suggestion
            while (!cancellationToken.IsCancellationRequested && (isInfiniteTimeSpan || (timeout >= TimeSpan.Zero)))
```
Otherwise, specifying `TimeSpan.Zero` will always result in `null`, which would be unexpected. Normally, specifying `TimeSpan.Zero` means that you want to "check the thing but don't wait for it if it's not ready.".